engine/gamesys/src/gamesys/components/comp_collection_proxy.cpp

```diff
@@ -166,14 +175,48 @@ namespace dmGameSystem
             dmhash_t component_instance_id = dmGameObject::GetIdentifier(c->m_Instance);
             if (component_instance_id == instanceId && c->m_ComponentIndex == index)
             {
-                comp_url_hash = c->m_Resource->m_UrlHash;
+                comp_url_hash = dmHashString64(GetCollectionResorcePath(c));
                 break;
             }
         }
 
         return comp_url_hash;
     }
 
+    SetCollectionForProxyPathResult CollectionProxySetCollectionPath(const HCollectionProxyWorld world, HCollectionProxyComponent component, const char* path)
+    {
+        CollectionProxyComponent* proxy = (CollectionProxyComponent*)component;
+
+        if (proxy->m_Loading)
+        {
+            return SET_COLLECTION_PATH_RESULT_COLLECTION_LOADING;
+        }
+        if (proxy->m_Collection)
+        {
+            return SET_COLLECTION_PATH_RESULT_COLLECTION_ALREADY_LOADED;
+        }
+        if (!proxy->m_Resource->m_DDF->m_Exclude)
+        {
+            return SET_COLLECTION_PATH_RESULT_COLLECTION_NOT_EXCLUDED;
+        }
+
+        if (proxy->m_CollectionResPath)
+        {
+            free(proxy->m_CollectionResPath);
+        }
+        
+        if (!path)
```

Comment:
I think we usually do the opposite,
if (thing)
{
}
else
{
    // !thing
}